drizzlepac/haputils/deconvolve_utils.py

```diff
@@ -131,127 +144,252 @@ def fft_deconv_img(img, psf, freq_limit=0.95, sigma=None):
+            for f in filter_list[2:]:
                 if f.lower().endswith(bp):
-                    filter_list = [f]
+                    kw_vals += [f.lower()]
+                    found_filter = True
                     break
-    # create full psf designation we are looking for:
-    # instr, det, filter
-    kw_vals += filter_list
-    # remove duplicate entries...
-    kw_vals = list(dict.fromkeys(kw_vals))
 
-    if kw_vals[0] is None:
-        log.error("No valid keywords found.")
-        raise ValueError
+    if not found_filter:
+        kw_vals += ['clear']
+    # The result at this point will be:
+    #  kw_vals = [instrument, detector, selected filter]
+    return kw_vals
 
-    if path_root is None:
-        path_root = os.path.split(os.path.dirname(__file__))[0]
-        for psf_path in PSF_PATH:
-            path_root = os.path.join(path_root, psf_path)
+def convert_library_psf(calimg, drzimg, psfs,
+                        total_flux=100000.0,
+                        pixfrac=1.0,
+                        clean_psfs=True):
+    """Drizzle library PSFs to match science image. """
 
-    path_root = os.path.join(path_root, kw_vals[0], kw_vals[1])
+    psf_flt_names = [_create_input_psf(psfname, calimg, total_flux) for psfname in psfs]
+
+    # Insure only final drizzle step is run
+    drizzle_pars = {}
+    drizzle_pars["build"] = True
+    drizzle_pars['context'] = False
+    drizzle_pars['preserve'] = False
+    drizzle_pars['clean'] = True
+    drizzle_pars['in_memory'] = True
+    drizzle_pars["resetbits"] = 0
 
-    psf_name = os.path.join(path_root, "{}.fits".format("_".join(kw_vals)))
+    drizzle_pars["static"] = False
+    drizzle_pars["skysub"] = False
+    drizzle_pars["driz_separate"] = False
+    drizzle_pars["median"] = False
+    drizzle_pars["blot"] = False
+    drizzle_pars["driz_cr"] = False
+    drizzle_pars["driz_combine"] = True
+    drizzle_pars['final_wcs'] = True
+    drizzle_pars['final_fillval'] = 0.0
+    drizzle_pars['final_pixfrac'] = pixfrac
+    drizzle_pars["final_refimage"] = "{}[1]".format(drzimg)
 
-    print('Looking for Library PSF {}'.format(psf_name))
-    log.debug('Looking for Library PSF {}'.format(psf_name))
+    psf_drz_name = psf_flt_names[0].replace('_flt.fits', '')  # astrodrizzle will add suffix
+    psf_drz_output = "{}_drz.fits".format(psf_drz_name)
 
-    if not os.path.exists(psf_name):
-        log.error('No PSF found for keywords {} \n   with values of {}'.format(INSTR_KWS, kw_vals))
-        raise ValueError
+    # Drizzle PSF FLT file to match orientation and plate scale of drizzled science (total detection) image
+    astrodrizzle.AstroDrizzle(input=psf_flt_names,
+                              output=psf_drz_name,
+                              **drizzle_pars)
 
-    log.info("Using Library PSF: {}".format(os.path.basename(psf_name)))
+    if clean_psfs:
+        # clean up intermediate files
+        for psf_name in psf_flt_names:
+            os.remove(psf_name)
```

Comment:
Cool. This was called out in HLA-493, but I see it getting cleaned now.

Reply:
Primary description at top edited to add a note about this change.